apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardImporter.java

```diff
@@ -98,22 +102,34 @@ protected void importGenesis(boolean onlyKeys) {
         genesisImporter.importGenesisJson(onlyKeys);
     }
 
-    public void importShard(String fileId, List<String> excludedTables) {
-        Objects.requireNonNull(fileId, "fileId is NULL");
+    public void importShard(ShardPresentData shardPresentData, List<String> excludedTables) {
+        Objects.requireNonNull(shardPresentData, "shardPresentData is NULL");
         Objects.requireNonNull(excludedTables, "excludedTables is NULL");
         // shard archive data has been downloaded at that point and stored (unpacked?) in configured folder
         String genesisTaskId = aplAppStatus.durableTaskStart("Shard data import", "Loading Genesis public accounts", true);
         log.debug("genesisTaskId = {}", genesisTaskId);
-        log.debug("Received shardFileId = '{}', lets map it to Location...", fileId);
-        Path zipInFolder = downloadableFilesManager.mapFileIdToLocalPath(fileId).toAbsolutePath();
-        log.debug("Try unpack file name '{}'", zipInFolder);
+        log.debug("Received shardPresentData = '{}', lets map all data to Location(s)...", shardPresentData);
+        Path zipInFolder = downloadableFilesManager.mapFileIdToLocalPath(shardPresentData.getShardFileId()).toAbsolutePath();
+        log.debug("Try unpack main shard file name '{}'", zipInFolder);
         boolean unpackResult = zipComponent.extract(zipInFolder.toString(), csvImporter.getDataExportPath().toString());
-        log.debug("Zip is unpacked = {}", unpackResult);
+        log.debug("Main shard Zip is unpacked = {}", unpackResult);
         if (!unpackResult) {
-            log.error("Node has encountered serious error and can't import ZIP with shard data. " +
-                    "Somethings wrong with zipped file =\n'{}'\n >>> STOPPING node process....", zipInFolder);
-            aplAppStatus.durableTaskFinished(genesisTaskId, true, "Shard data import");
-            throw new ShardArchiveProcessingException("Zip file can't be extracted, result = '" + unpackResult + "' : " + zipInFolder.toString());
+            logErrorAndThrowException(shardPresentData, genesisTaskId, zipInFolder, unpackResult);
+            return;
+        }
+        // unzip additional files
+        if (shardPresentData.getAdditionalFileIDs() != null && shardPresentData.getAdditionalFileIDs().size() > 0) {
```

Comment:
importShard seems to be a big one. What do you think of separating "unzip additional files" into a new method and call it here?

Reply:
done, extracted private method for unzip